src/common/settings.cpp

```diff
@@ -31,6 +31,9 @@ enum class Source {
    CONFIG_FILE_DEFAULT_SECTION
 };
 
+// Json object key for the auto-generated warning comment
+const std::string SETTINGS_WARN_MSG_KEY{"_warning_"};
```

Comment:
In commit "init: settings, do not load auto-generated warning msg" (987a1b51eeb72c7fcb085470817a4fe85fcc3c7c)

Not sure, but maybe this can be changed to std::string_view to avoid the need to allocate memory before main() runs.

Reply:
> In commit "init: settings, do not load auto-generated warning msg" ([987a1b5](https://github.com/bitcoin/bitcoin/commit/987a1b51eeb72c7fcb085470817a4fe85fcc3c7c))
> 
> Not sure, but maybe this can be changed to std::string_view to avoid the need to allocate memory before main() runs.

As the settings map take strings for keys, wouldn't this change mean that we would need to static cast string_view to string to not create a copy during the push/erase methods call?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/29301#discussion_r1465379877

> As the settings map take strings for keys, wouldn't this change mean that we would need to static cast string_view to string to not create a copy during the push/erase methods call?

The idea is that just that it is noticeable when programs are slow to start, so you usually want to reduce the amount of code that needs to run before main(), even if it means code that runs later but is less sensitive might be slower. In this case there is not even a need for a tradeoff, though since univalue could take string_view arguments. Just explaining the suggestion though, this is not important

Reply:
Sure, sounds good @ryanofsky. Thanks for coming back to this!